src/VisualStudio/Core/Def/Implementation/ProjectSystem/VisualStudioProjectTracker.cs

```diff
@@ -827,9 +836,13 @@ private void OutputToOutputWindow(string message)
                     continue;
                 }
 
-                projectContext.AddMetadataReference(path, reference.Properties);
+                if (addedReferencePaths.Add(path))
```

Comment:
You could have two different aliases for the same reference. I hope nobody in practice ever does that.